temporal-sdk/src/test/java/io/temporal/internal/sync/WorkflowInternalQueueTest.java

```diff
@@ -386,4 +386,26 @@ public void testMap() throws Throwable {
     trace.setExpected(expected);
     r.close();
   }
+
+  @Test
+  public void testQueueOrder() throws Throwable {
+    DeterministicRunner r =
+        DeterministicRunner.newRunner(
+            () -> {
+              WorkflowQueue<Integer> queue = WorkflowInternal.newQueue(3);
+              queue.put(1);
+              queue.put(2);
+              queue.put(3);
+              trace.add(queue.take().toString());
```

Comment:
I believe we use Tracer mostly for testing of threads' order and behavior. We can just use a regular collection here to persist the elements and compare the collections at the end of the test for simplicity and to don't depend on Tracer where we don't really need it. 
Not a big issue though, I will leave the PR open for some time if you want to refactor the test just to use a regular collection.

Reply:
Totally makes sense, will update it now.

Reply:
Updated.

One thing I've realized - I worry if this fix might break something? E.g. can it happen that upgrading Temporal SDK would introduce a non-deterministic update to the workflow?